pulp_container/app/schema_convert.py

```diff
@@ -54,7 +52,14 @@ def _convert_schema(self, manifest):
             )
         except ValueError:
             raise RuntimeError()
-        return converter.convert()
+
+        converted_schema, schema_with_signature = converter.convert()
```

Comment:
Doesn't convert return three values?
I guess i am confused by `Schema2toSchema1ConverterWrapper` vs `Schema2toSchema1Converter`.

Reply:
No, it returns two values. See the method `convert` of the class `Schema2toSchema1Converter`. The method `_convert_schema` of the class `Schema2toSchema1ConverterWrapper` returns three values.

Reply:
Can we simplify this in a follow up PR? It seems a bit unnecessary to write a wrapper around a class that isn't used elsewhere whilst both are under our control.
But, please don't mix such a simplification with a change in behavior, but make it two PR's.

Reply:
Will do.

Reply:
After a thorough examination, I am not going to do this. The wrapper class serves as a manager that executes the conversion only when it is required. Moving the part that determines the conversion to the class `Schema2toSchema1Converter` will violate the single responsibility principle. Furthermore, it will make the unit testing more complicated without additional monkey patching.

Maybe, we could just rename the class `Schema2toSchema1ConverterWrapper` to `Schema2toSchema1ConverterHandler` or `Schema2toSchema1ConverterManager`?